transaction/core/src/onetime_keys.rs

```diff
@@ -317,16 +299,14 @@ mod tests {
 
         assert_eq!(D_prime, *recipient.spend_public_key()); // D_7
 
-        // view_key_matches_output should return true.
-        let view_key = ViewKey::new(
-            account.view_private_key().clone(),   //a
-            recipient.spend_public_key().clone(), // D_7
+        assert_eq!(
```

Comment:
`view_key_matches_output` does not actually do view key matching (very confusingly, which is why i want to remove it)

it checks if the subaddress corresponding to the `ViewKey` owns the `TxOut`

So that's what it gets replaced with in the tests, to avoid changing the semantics of the tests